docs/2024/puzzles/day18.md

```diff
@@ -1,11 +1,181 @@
 import Solver from "../../../../../website/src/components/Solver.js"
 
 # Day 18: RAM Run
+by [@TheDrawingCoder-Gamer](https://github.com/TheDrawingCoder-Gamer)
 
 ## Puzzle description
 
 https://adventofcode.com/2024/day/18
 
+## Solution Summary
+
+1. Parse input into a List of points
+2. For part 1, take the first 1024 points and path find through them
+3. For part 2, find the tipping point where pathfinding can't be done anymore
+
+## Part 1
+
+First, let's make the Vec2i class for our points:
+
+```scala
+case class Vec2i(x: Int, y: Int):
+  def cardinalNeighbors: List[Vec2i] =
+    List(Vec2i(x - 1, y), Vec2i(x + 1, y), Vec2i(x, y - 1), Vec2i(x, y + 1))
+
+  def isContainedIn(w: Int, h: Int): Boolean =
+    x >= 0 && x < w && y >= 0 && y < h
+```
+
+Then let's parse all our points. (because the example is so different from the actual input, there is special casing for it.)
+
+```scala
+def parse(str: String): (Int, Int, List[Vec2i]) =
+  val ps = str.linesIterator.map:
+    case s"$x,$y" => Vec2i(x.toInt, y.toInt)
+  .toList
+  (if ps.length == 25 then 7 else 71, if ps.length == 25 then 12 else 1024, ps)
+```
+
+For both parts, we'll need a path finding function. While I would usually use a grid, because of part 2 it's easier to pathfind using a Set.
```

Comment:
```suggestion
For both parts, we'll need a path finding function. While I would usually use a grid, because of part 2 it's easier to pathfind using a `Set`.
```